src/ser/mod.rs

```diff
@@ -363,11 +388,17 @@ impl<'a, W: io::Write> ser::Serializer for &'a mut Serializer<W> {
 
     fn serialize_f32(self, v: f32) -> Result<()> {
         write!(self.output, "{}", v)?;
+        if self.decimal_floats() && v - v.floor() == 0.0 {
```

Comment:
why isn't this written as `v == v.floor()`?

Reply:
No reason. Your representation is clearly better :smile: